packages/core/upload/admin/src/components/FolderList/FolderList.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { stringify } from 'qs';
+import { useHistory, useLocation } from 'react-router-dom';
+import { Box } from '@strapi/design-system/Box';
+import { KeyboardNavigable } from '@strapi/design-system/KeyboardNavigable';
+import { Flex } from '@strapi/design-system/Flex';
+import { IconButton } from '@strapi/design-system/IconButton';
+import { Typography } from '@strapi/design-system/Typography';
+import { VisuallyHidden } from '@strapi/design-system/VisuallyHidden';
+import { useQueryParams } from '@strapi/helper-plugin';
+import Pencil from '@strapi/icons/Pencil';
+
+import { FolderCard, FolderCardBody, FolderCardCheckbox, FolderCardLink } from '../FolderCard';
+import { GridColumnSize } from '../../constants';
+
+const GridLayout = styled(Box)`
+  display: grid;
+  grid-template-columns: repeat(
+    auto-fit,
+    minmax(${({ size }) => `${GridColumnSize[size]}px`}, 1fr)
+  );
+  grid-gap: ${({ theme }) => theme.spaces[4]};
+`;
+
+const CardTitle = styled(Typography).attrs({
+  ellipsis: true,
+  fontWeight: 'semiBold',
+  textColor: 'neutral800',
+  variant: 'omega',
+})`
+  max-width: 100%;
+`;
+
+export const FolderList = ({
+  title,
+  folders,
+  size,
+  onSelectFolder,
+  onEditFolder,
+  selectedFolders,
+}) => {
+  const history = useHistory();
+  const { pathname } = useLocation();
+  const [{ query }] = useQueryParams();
```

Comment:
While I like the idea of a unified hook, I'm not sure I think it is a good idea. Why? Because `react-router` brings it's own implementation of [`useParams()`](https://reactrouter.com/docs/en/v6/hooks/use-params) and I'd be afraid that it leads to confusion because we mix up our own implementation with `react-router` (`useLocation()`). 

However: it might be worth checking, now that there is a react-router compat (which makes v6 compatible with the v5 API we use) - to check if we can upgrade it and then replace `useQueryParams()` with `useParams()` ...

I think both options need to be discussed in a FE sync and added as part of another PR. What do you think?

Reply:
Good point! Let's keep this topic for the next front end sync